src/_http/http2Adapter.js

```diff
@@ -279,4 +325,46 @@ Http2Adapter.prototype.execute = function(options) {
   })
 }
 
+/**
+ * Moves to the closed state, cleans up ongoing HTTP2 sessions if any.
+ *
+ * @param {?object} opts Close options.
+ * @param {?boolean} opts.force Whether to force resources clean up.
+ * @returns {Promise<void>}
+ */
+Http2Adapter.prototype.close = function(opts) {
+  opts = opts || {}
+
+  this._closed = true
+
+  var noop = function() {}
+
+  return Promise.all(
+    Object.values(this._sessionMap).map(function(sessionInterface) {
+      return sessionInterface.close(opts.force)
+    })
+  ).then(noop)
```

Comment:
I'm curious why the noop is needed?

Reply:
@retwere It forces returning `Promise<void>` instead of `Promise<undefined[]>` to be consistent and conform to JS doc definition, otherwise:
```typescript
Promise.all([Promise.resolve()]) // Returns Promise<undefined[]>
```
If we used async/await, we could just "await" it and return nothing:
```typescript
  ...
  await Promise.all(...)
}
```